__init__.py

```diff
@@ -140,6 +115,15 @@ def _find_entity(self, entity, domains):
                               "dev_name": entity})
         return ha_entity
 
+    # Routine for entiti availibility check
+    def _check_availability(self, ha_entity):
+        self.log.debug(ha_entity['state'])
+        if ha_entity['state'] == 'unavailable':
```

Comment:
What is the outcome of this function if ha_entity['state'] does not exist? If the ha_entity is empty for instance?

Reply:
If entity not exist, You will not even call this function

`if not ha_entity or not self._check_availability(ha_entity):`